### PR TITLE
Update Cluster API Visualizer chart to v1.0.0

### PR DESCRIPTION
**Pull request by Jont828**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: hack/observability/cluster-api-visualizer/chart/templates/deployment.yaml [hunk 1]
```diff
@@ -16,18 +17,8 @@ spec:
     spec:
       containers:
       - name: capi-visualizer
-        image: "{{ .Values.image.repository }}/{{ .Values.image.name }}:{{ .Values.image.tag }}"
+        image: "{{ .Values.image.repository }}/{{ .Values.image.name }}:{{ default .Chart.AppVersion .Values.image.tag }}"
         imagePullPolicy: {{ .Values.image.pullPolicy }}
         ports:
         - containerPort: 8081
-        env:
-        - name: KUBECONFIG
```
Comment (CecileRobertMichon) [inline hack/observability/cluster-api-visualizer/chart/templates/deployment.yaml:24]: ❤️

File: hack/observability/cluster-api-visualizer/fetch-visualizer-resources.sh [hunk 2]
```diff
@@ -18,12 +18,12 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-CHART_RELEASE=${CHART_RELEASE:-0.1.0}
+CHART_RELEASE=${CHART_RELEASE:-1.0.0}
```
Comment (CecileRobertMichon) [inline hack/observability/cluster-api-visualizer/fetch-visualizer-resources.sh:21]: are you planning on cutting a github v1.0.0 release along with this chart release https://github.com/Jont828/cluster-api-visualizer? 

That would be a cool announcement to make in CAPI channels

Comment (Jont828) [reply hack/observability/cluster-api-visualizer/fetch-visualizer-resources.sh:21]: I haven't really thought about that at the moment but I can look into it!

**Review by mboersma — COMMENTED** [review]
/lgtm

I've been testing this out locally and I haven't seen any problems under `make tilt-up`. The visualizer is quite useful, I'm finding it easier to locate certain errors there than in tilt logs.

File: hack/observability/cluster-api-visualizer/chart/templates/clusterrole.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: capi-visualizer
+rules:
+# TODO: Consider restricting the rights to the relevant CAPI apiGroups.
```
Comment (CecileRobertMichon) [inline hack/observability/cluster-api-visualizer/chart/templates/clusterrole.yaml:6]: Can we open a good-first-issue for this?

Comment (Jont828) [reply hack/observability/cluster-api-visualizer/chart/templates/clusterrole.yaml:6]: Sure thing!

**Pull request merged**